### PR TITLE
Move the clipboard handling into the core library

### PR DESCRIPTION
**Pull request by tronical**
That removes two more functions from the backend interface.

### CODE REVIEW

File: internal/core/Cargo.toml [hunk 1]
```diff
@@ -83,6 +85,8 @@ resvg = { version= "0.23", optional = true, default-features = false }
 usvg = { version= "0.23", optional = true, default-features = false, features = ["text"] }
 tiny-skia = { version= "0.6", optional = true, default-features = false }
 
+copypasta = { version = "0.8.1", optional = true, default-features = false, features = ["x11"] }
```
Comment (ogoffart) [inline internal/core/Cargo.toml:88]: oh, x11 and wayland again....

File: internal/core/clipboard.rs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+/// Returns the text that is stored in the system clipboard, if any.
+pub fn clipboard_text() -> Option<String> {
+    #[cfg(feature = "clipboard")]
+    {
+        use copypasta::ClipboardProvider;
+        return CLIPBOARD.with(|clipboard| clipboard.borrow_mut().get_contents().ok());
+    }
+    #[cfg(not(feature = "clipboard"))]
+    None
```
Comment (ogoffart) [inline internal/core/clipboard.rs:50]: Could we not at least store it on a local string so it works on all platform. even Web and MCU

File: internal/core/Cargo.toml [hunk 3]
```diff
@@ -23,14 +23,16 @@ libm = ["num-traits/libm", "euclid/libm"]
...
-std = ["euclid/std", "once_cell/std", "scoped-tls-hkt", "lyon_path", "lyon_algorithms", "lyon_geom", "lyon_svg", "instant", "image-decoders", "svg"]
+std = ["euclid/std", "once_cell/std", "scoped-tls-hkt", "lyon_path", "lyon_algorithms", "lyon_geom", "lyon_svg", "instant", "image-decoders", "svg", "clipboard"]
 # Unsafe feature meaning that there is only one core running and all thread_local are static.
 # You can only enable this feature if you are sure that any API of this crate is only called
 # from a single core, and not in a interrupt or signal handler.
 unsafe_single_core = []
 
 text_layout = []
 
+clipboard = ["copypasta"]
```
Comment (ogoffart) [inline internal/core/Cargo.toml:34]: Do we really need a feature for that. Can we not simply use "copypasta" as the feature.

File: internal/core/Cargo.toml (hunk 1 above)
Comment (tronical) [reply internal/core/Cargo.toml:88]: Should be fine I think. We don’t use the Wayland but and the x11 features compiles also on non-x11

**Comment by tronical** [comment]
Abandoning this - the clipboard works via the windowing system, which should re-use an existing connection (instead of creating new ones). Will work a different way via the Window instead of Backend.

**Pull request closed without merging**